data-access-layer/datastore/database.go

```diff
@@ -281,17 +281,26 @@ func getTx(db *sql.DB, orgId string, isMultitenant bool) (*sql.Tx, error) {
 Validates tenancy information and returns a transaction with right dbRole.
 */
 func (database *RelationalDb) validateAndGetTx(ctx context.Context, tableName string, record Record, _ string) (*sql.Tx, error) {
-	err, orgId, dbRole, isMultitenant := database.getTenantInfoFromCtx(ctx, tableName)
+	err, orgId, dbRole, isDbRoleTenantSpecific := database.getTenantInfoFromCtx(ctx, tableName)
 	if err != nil {
 		return nil, err
 	}
 
-	if err = database.authorizer.IsOperationAllowed(ctx, tableName, record); err != nil {
-		logger.Errorf("Operation not allowed for the following reason: %s", err.Error())
-		return nil, err
+	/*
+		If the DB role is tenant-specific (TENANT_READER or TENANT_WRITER) and the table is multi-tenant,
+		make sure that the record being inserted/modified/updated/deleted/queried belongs to the user's org.
+		If operation is SELECT but no specific tenant's data is being queried (e.g., FindAll() was called), allow the operation to proceed.
+	*/
+	if dbRole.isDbRoleTenantSpecific() && IsMultitenant(record, tableName) {
+		orgIdCol := getField(ORG_ID_COLUMN_NAME, record)
+		if orgIdCol != "" && orgIdCol != orgId {
```

Comment:
should we fail here for non SELECT operations here based on line #292?

Reply:
INSERT, UPDATE, DELETE will fail if you're trying to insert/update/delete a record of another tenant.
SELECT will also fail if you're trying to read a single record of another tenant.
Only if SELECT is trying to read all tenants' data will this if statement have no effect.

Reply:
so, non-SELECT operations are already failing if you're trying to modifying another tenant's data